views/components/form/resource_creation_form.erb

```diff
@@ -64,20 +64,19 @@ let option_dirty = <%==JSON.generate(form_elements.map { it[:name] }.each_with_o
 
       opt_attrs = {}
       opt_attrs[:disabled] = "disabled" unless parents_selected
-      opt_attrs[:checked] = "checked" if checked
+      opt_attrs[:checked] = checked
 
       [opt_val, opt_text, opt_classes, opt_attrs]
     end if has_option
 
     if has_option && selected_options[name].nil?
       options.find{ |_, _, _, opt_attrs| opt_attrs[:disabled].nil? }&.tap do |opt_val, _, _, opt_attrs|
-        opt_attrs[:checked] = "checked"
+        opt_attrs[:checked] = true
         selected_options[name] = opt_val
       end
     end
 
-    attributes = {}
-    attributes[:required] = "" if required
+    attributes = {required:}
```

Comment:
Do we need to add
```
attributes[:required] = true if required
```

, so that html_attrs wouldn't ignore this attribute?

Reply:
`attributes = {required:}` is short for `attributes = {required: required}` (hash value omission).  So this uses the value of the `required` variable.  If `required` is true, this would be `attributes = {required: true}`. Similar if required is `false` or `nil`.  One case where this has different behavior is if `required` was `"true"`.  However, I don't think we have those cases (if we did, the custom HTML validation added would catch it.

Reply:
Ah yes. I forgot about hash value omission :+1: